models/src/main/scala/coop/rchain/models/ParMap.scala

```diff
@@ -8,15 +8,12 @@ import coop.rchain.models.rholang.implicits._
 case class ParMap(ps: SortedParMap, connectiveUsed: Boolean, locallyFree: Coeval[BitSet]) {
 
   override def equals(o: scala.Any): Boolean = o match {
-    case parMap: ParMap =>
-      this.ps == parMap.ps && this.connectiveUsed == parMap.connectiveUsed && this.locallyFree.value == parMap.locallyFree.value
-    case _ => false
+    case parMap: ParMap => this.ps == parMap.ps
```

Comment:
I've seen the commit message, but it's not clear to me:

> locallyFree was always redundant in patterns because they were never going to be
executed or be matched. Now it is the top-level locallyFree cache.

I assume this changed when introducing bound variables in patterns? E.g. in
```
for (@x <- @"x") {
  match 7 {
     =x => P
}
```
(thx @goral09 for this!)

The commit message should include this.

Is the top-level locallyFree cache introduced by this commit?
Why is it needed?

> Because we use equality comparisons in matches
without connectives, we need to ignore locallyFree when doing equality
comparisons. Hence locallyFree. 

An example of a equality-compared match without connectives would help understand the rationale for this change.

I'm not sure what "Hence locallyFree" means in this context.

> HasLocallyFree.locallyFree relies on the cached information, and therefore will
only return the correct result if asked about the top-level.

I'm not sure why we need that caching. Please add the reason and/or at least the consequence quoted above into `HasLocallyFree`'s scaladoc.

Reply:
I added comments to HasLocallyFree